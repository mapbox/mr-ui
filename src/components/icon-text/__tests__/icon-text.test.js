import React from 'react';
import { shallow } from 'enzyme';
import { testCases } from './icon-text-test-cases';

let testCase;
let wrapper;

describe(testCases.iconBefore.description, () => {
  beforeEach(() => {
    testCase = testCases.iconBefore;
    wrapper = shallow(React.createElement(testCase.component, testCase.props));
  });

  test('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe(testCases.iconAfter.description, () => {
  beforeEach(() => {
    testCase = testCases.iconAfter;
    wrapper = shallow(React.createElement(testCase.component, testCase.props));
  });

  test('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe(testCases.largeSpacing.description, () => {
  beforeEach(() => {
    testCase = testCases.largeSpacing;
    wrapper = shallow(React.createElement(testCase.component, testCase.props));
  });

  test('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe(testCases.bothIcons.description, () => {
  beforeEach(() => {
    testCase = testCases.bothIcons;
    wrapper = shallow(React.createElement(testCase.component, testCase.props));
  });

  test('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
