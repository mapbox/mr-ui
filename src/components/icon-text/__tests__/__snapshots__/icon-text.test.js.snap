// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon after renders as expected 1`] = `
<span
  className="flex-parent flex-parent--center-cross"
>
  <span
    className="flex-child"
  >
    Next
  </span>
  <span
    className="flex-child ml3"
  >
    <Icon
      inline={false}
      name="chevron-right"
      passthroughProps={Object {}}
      size={18}
    />
  </span>
</span>
`;

exports[`icon before renders as expected 1`] = `
<span
  className="flex-parent flex-parent--center-cross"
>
  <span
    className="flex-child mr3"
  >
    <Icon
      inline={false}
      name="clipboard"
      passthroughProps={Object {}}
      size={18}
    />
  </span>
  <span
    className="flex-child"
  >
    Copy
  </span>
</span>
`;

exports[`icons on both sides, including a non-string one renders as expected 1`] = `
<span
  className="flex-parent flex-parent--center-cross"
>
  <span
    className="flex-child mr3"
  >
    <Icon
      inline={false}
      name="check"
      passthroughProps={Object {}}
      size={18}
    />
  </span>
  <span
    className="flex-child"
  >
    What
  </span>
  <span
    className="flex-child ml3"
  >
    <Icon
      inline={false}
      name="github"
      passthroughProps={Object {}}
      size={18}
    />
  </span>
</span>
`;

exports[`large gap renders as expected 1`] = `
<span
  className="flex-parent flex-parent--center-cross"
>
  <span
    className="flex-child mr6"
  >
    <Icon
      inline={false}
      name="check"
      passthroughProps={Object {}}
      size={18}
    />
  </span>
  <span
    className="flex-child"
  >
    Done
  </span>
</span>
`;
